src/frontend/efiling-demo/src/components/page/home/Home.js

```diff
@@ -139,6 +121,100 @@ export default function Home({ page: { header } }) {
   const [submitBtnEnabled, setSubmitBtnEnabled] = useState(true);
   const [showLoader, setShowLoader] = useState(false);
 
+  const generateTable = (file, data) => [
+    {
+      name: (
+        <div style={{ width: "55%" }}>
+          <span
+            onKeyDown={() => {}}
+            role="button"
+            tabIndex={0}
+            className="file-href"
+            onClick={() => {}}
+            data-test-id="uploaded-file"
+          >
+            {file.name}
+          </span>
+        </div>
+      ),
+      value: (
+        <div className="d-flex justify-content-around bcgov-row">
+          {data.map((option) => (
+            <div className="d-flex align-items-center">
+              <h5 className="ml-5 mr-2 mt-3">{option.name}</h5>
+              {option.value}
+            </div>
+          ))}
+        </div>
+      ),
+    },
+  ];
+
+  const generateTableData = (file) => {
+    file.data = {};
+    const data = [
+      {
+        name: "Type:",
+        value: (
+          <Dropdown
+            items={docTypeList}
+            onSelect={(e) => {
+              file.data = { ...file.data, type: e.split(/-(.+)/)[0].trim() };
```

Comment:
You could make your life easier by jsoning your file like 
```json
[
{
  "key": "ABP",
  "value": "Abandonment/Settlement (Partial)"
}, ... ]
```
this would remove the need of the regex ;-)

Reply:
That is how i inititally tried to implement it, however I switched to more of a list structure because the shared components <Dropdown /> and <DisplayBox /> are not quite set up to accept key value pairs like this, and I wanted to keep things consistent. However if its a deal breaker I can definitely give it another go.

Reply:
I have converted the typeList to an array of objects. I also adjusted Home.js to only display the type in the dropdown